### PR TITLE
Guardian Updates

### PR DESCRIPTION
**Pull request by ghost**
Ghosts can now read Guardian/Summoner communications.

Removes the scout type.

The ranged type can now toggle between ranged properties/scout properties while recalled. It can also deploy invisible traps that will alert it when a mob moves over them.

Fire type does 15 damage a hit again, gets the PASSMOB flag so people on harm intent aren't immune to being set on fire (via Crossed)

Fixes a typo.

### CODE REVIEW

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 1]
```diff
@@ -280,10 +274,79 @@
...
+			range = 13
+			incorporeal_move = 0
+			src << "<span class='danger'><B>You switch to combat mode.</span></B>"
+		else
+			ranged = 0
+			melee_damage_lower = 0
+			melee_damage_upper = 0
+			alpha = 60
+			range = 255
+			incorporeal_move = 0
```
Comment (Anonus) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: Shouldn't this be 1?

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 2]
```diff
@@ -280,10 +274,79 @@
...
+	set name = "Set Surveillance Trap"
+	set category = "Guardian"
+	set desc = "Set an invisible trap that will alert you when living creatures walk over it. Max of 5"
+	if(src.snares.len <6)
+		var/turf/snare_loc = get_turf(src.loc)
+		var/obj/item/effect/snare/S = new /obj/item/effect/snare(snare_loc)
+		S.spawner = src
+		S.name = "[get_area(snare_loc)] trap ([rand(1, 1000)])"
+		src.snares |= S
+		src << "<span class='danger'><B>Surveillance trap deployed!</span></B>"
```
Comment (Anonus) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: Should tell you the number of the trap, as well.

**Comment by PKPenguin321** [comment]
will you update the wiki with this or are you cool with me doing it again (once it's merged)

**Comment by KorPhaeron** [comment]
I'll try to remember to update it now that the basic formatting was all done by you (thanks)

**Comment by PKPenguin321** [comment]
i also have a few suggestions:
make ranged type's melee attack do significantly less damage, it should be almost entirely focused on its ranged attack
make fire type do burn instead of brute with it's melee attack because it's flavorful
make explosive type do less melee damage, it should be focused on sabotage but instead it's basically a better version of the fast type
give shield type some kind of weak attack/the forcewall spell/the ability to smash down walls, it's kind of underpowered considering healer type does its job of tanking damage better
make the healer go slightly faster so it has more utility in combat (see: escaping from combat)

i guess you can take these all with a grain of salt since you're the creator and you can do what you want, but they're something to consider

**Comment by KorPhaeron** [comment]
>make ranged type's melee attack do significantly less damage, it should be almost entirely focused on its ranged attack

I would but simple animal ranged targeting is wonky and they need a way to attack people on the ground

>make fire type do burn instead of brute with it's melee attack because it's flavorful

It's hands are on fire!

>make explosive type do less melee damage, it should be focused on sabotage but instead it's basically a better version of the fast type

Never seen fast type taken, I'm gonna scrap it and redo it. I want all of them to have some utlity+basic combat usefulness eventually

>give shield type some kind of weak attack/the forcewall spell/the ability to smash down walls, it's kind of underpowered considering healer type does its job of tanking damage better

Agreed. I was thinking of merging the two types actually.

>make the healer go slightly faster so it has more utility in combat (see: escaping from combat)

Problem would be you now have an unstunnable, unslowable mob running away spamming heal forever. I agree it's shitty being slow right now though. Hence the merging with shield to make a general tanky type (plus a forcewall probably).

**Pull request merged**